mobile/native/src/api.rs

```diff
@@ -217,3 +224,38 @@ pub fn get_seed_phrase() -> Result<SyncReturn<Vec<String>>> {
     let seed_phrase = ln_dlc::get_seed_phrase()?;
     Ok(SyncReturn(seed_phrase))
 }
+
+pub struct LightningInvoice {
+    pub description: String,
+    pub amount_sats: f64,
+    pub timestamp: u64,
+    pub payee: String,
+    pub expiry: u64,
+}
+
+pub fn decode_invoice(invoice: String) -> Result<LightningInvoice> {
+    anyhow::ensure!(!invoice.is_empty(), "received empty invoice");
+    let invoice = &Invoice::from_str(&invoice).context("Could not parse invoice string")?;
+    let description = match invoice.description() {
+        InvoiceDescription::Direct(direct) => direct.to_string(),
+        InvoiceDescription::Hash(_) => "".to_string(),
+    };
+    // can't use as_millis as the frb does not support u128
+    let timestamp = invoice
```

Comment:
What's the timestamp needed for?

Reply:
I just blindly copied this from the POC - will have another look :)

Reply:
In the poc we just displayed it - no other use. 
As far as I understand it's the expiry timestamp of the invoice. 
I would keep it for completion.

Reply:
> As far as I understand it's the expiry timestamp of the invoice.

I'm confused :) The expiry is 4 lines below.

Reply:
The timestamp is when the invoice has been created and the expiry defines the number of seconds until it is invalid.

The following example should make it clear. 
```
"timestamp": "1496314658",
"expiry": "3600",
```